source/gui/__init__.py

```diff
@@ -967,21 +969,14 @@ def onYesButton(self,evt):
 		log.debug("Usage stats gathering has been allowed")
 		config.conf['update']['askedAllowUsageStats']=True
 		config.conf['update']['allowUsageStats']=True
-		self.Destroy()
+		self.EndModal(wx.ID_YES)
 
 	def onNoButton(self,evt):
 		log.debug("Usage stats gathering has been disallowed")
 		config.conf['update']['askedAllowUsageStats']=True
 		config.conf['update']['allowUsageStats']=False
-		self.Destroy()
+		self.EndModal(wx.ID_NO)
 
 	def onLaterButton(self,evt):
 		log.debug("Usage stats gathering question has been deferred")
 		evt.Skip()

```

Comment:
I'm surprised that you don't need to call `EndModal` here? Could it be because of the `evt.Skip()`?

Reply:
Why are you calling `Skip()` on this?